### PR TITLE
[BUG] Fix Benchmarking duplicate estimator label collisions

### LINKED ISSUE #297: [BUG] Benchmarking collapses multiple estimators of the same class into one result row
## Describe the bug
`Benchmarking.run()` stores results in a dict keyed only by `estimator.__class__.__name__`, so benchmarking two estimators of the same class silently overwrites the first result with the second.

This means hyperparameter comparisons within the same estimator family are not represented correctly in the summary table.

Relevant code:
- `pyaptamer/benchmarking/_base.py` (`est_name = estimator.__class__.__name__` and `results[est_name] = est_scores`)

## Minimal reproduction
```python
import numpy as np
from sklearn.dummy import DummyClassifier
from sklearn.metrics import accuracy_score
from pyaptamer.benchmarking import Benchmarking

X = np.array([[0], [1], [0], [1], [0], [1], [0], [1]])
y = np.array([0, 1, 0, 1, 0, 1, 0, 1])

bench = Benchmarking(
    estimators=[
        DummyClassifier(strategy="most_frequent"),
        DummyClassifier(strategy="stratified", random_state=0),
    ],
    metrics=[accuracy_score],
    X=X,
    y=y,
    cv=2,
)

print(bench.run())
```

Current behavior:
- only one `DummyClassifier` row is returned

Expected behavior:
- both estimators should appear in the results under distinct labels
- distinct estimator classes should keep their original class-name labels

## Proposed fix
Generate stable display names for estimators before running CV, and disambiguate duplicate class names with a deterministic suffix such as `[1]`, `[2]`, etc.

### PR DESCRIPTION
**Pull request by ashum9**
Closes #297.

## What this changes
- disambiguates duplicate estimator class names in `Benchmarking` results with deterministic 1-based suffixes
- preserves the original class name when there is only one estimator of that class
- adds a regression test proving two `DummyClassifier` instances no longer overwrite one another
- adds a regression test proving distinct estimator classes keep their original labels

## Validation
- `pytest -q pyaptamer/benchmarking/tests/test_benchmarking_core.py`

### CODE REVIEW

File: pyaptamer/benchmarking/tests/test_benchmarking_core.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        ],
+        metrics=[accuracy_score],
+        X=X,
+        y=y,
+        cv=2,
+    )
+
+    summary = bench.run()
+
+    assert ("DummyClassifier[1]", "accuracy_score") in summary.index
```
Comment (siddharth7113) [inline pyaptamer/benchmarking/tests/test_benchmarking_core.py]: this notation wouldn't be right, and could be confusing in downstream usage, user would have to make a call something like this `df.loc["DummyClassifier[1]"]` , I would suggest to index along with strategy something to make it either like `DummyClassifer_1` or based on Pattern should be fine as well.

**Review by siddharth7113 — CHANGES_REQUESTED** [review]
left the comment to change the suffix structure, otherwise looks good.

**Comment by ashum9** [comment]
@siddharth7113 Thanks for the feedback. I updated the duplicate-estimator label format to use underscore suffixes instead of brackets.

Now duplicate estimators of the same class are labeled deterministically as `DummyClassifier_1`, `DummyClassifier_2` (1-based, only when needed), while unique estimator classes keep their original class name.

Updated the regression tests accordingly; `pytest -q pyaptamer/benchmarking/tests/test_benchmarking_core.py` passes.

**Comment by siddharth7113** [comment]
Please follow PR guidelines and make sure to install and run `pre-commit` , the current CI checks are failing.

**Comment by siddharth7113** [comment]
cc @satvshr , for your feedback

File: pyaptamer/benchmarking/tests/test_benchmarking_core.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    )
+
+    summary = bench.run()
+
+    assert ("DummyClassifier_1", "accuracy_score") in summary.index
+    assert ("DummyClassifier_2", "accuracy_score") in summary.index
+    assert len(summary) == 2
+
+
+def test_benchmarking_preserves_unique_estimator_names():
```
Comment (satvshr) [inline pyaptamer/benchmarking/tests/test_benchmarking_core.py:31]: Was this also a problem which you solved or are you adding a random test
